engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/BeanMetaDataImpl.java

```diff
@@ -175,7 +176,15 @@ public BeanMetaDataImpl(Class<T> beanClass,
 				cascadedProperties.add( propertyMetaData );
 			}
 			else {
-				allMetaConstraints.addAll( propertyMetaData.getTypeArgumentsConstraints() );
+				Set<MetaConstraint<?>> typeConstraints = propertyMetaData.getTypeArgumentsConstraints().stream()
```

Comment:
My question was more: is this special casing specific to this place or should we move it a bit sooner to get PropertyMetaData return these transformed type parameter constraints?